fdbserver/workloads/ConsistencyCheck.actor.cpp

```diff
@@ -1325,7 +1370,7 @@ struct ConsistencyCheckWorkload : TestWorkload {
 						// Get the min version of the storage servers
 						Version version = wait(self->getVersion(cx, self));
 
-						state GetKeyValuesRequest req;
+						state GetKeyValuesStreamRequest req;
```

Comment:
Is the rate limiter being applied during a shard read or after?  It's especially important to apply it during the read since we're using streaming reads here.

Also, it's unclear to me when reading via streaming from a specific storage interface how much read ahead will be done.  Streaming read via a native client will send up to 20 1mb streaming read requests at once across the storage team, but I'm not sure what will happen here - it is sending the same type of request as the 1mb streaming read requests but with a larger range I think.  Trying to read a whole shard via streaming read would probably not be good, but rate limiting on the receiving side might be enough to limit the remote read ahead.

Reply:
The rate limiter is used after the read. The receiving side rate limit is in affect. Also we don't use parallelism in the streaming read here. Only the read ahead apart is leveraged. I will have Evan also review this part.